pkg/scheduler/internal/cache/cache.go

```diff
@@ -585,6 +585,10 @@ func (cache *schedulerCache) AddCSINode(csiNode *storagev1beta1.CSINode) error {
 }
 
 func (cache *schedulerCache) UpdateCSINode(oldCSINode, newCSINode *storagev1beta1.CSINode) error {
+	if oldCSINode.Name != newCSINode.Name {
```

Comment:
AFAIK the API server validation mechanism prevents change of name of objects by default. So, this check may not be needed.

Reply:
If that is the case, we should drop the oldCSINode parameter.

Reply:
For update API, it makes sense to have both old and new objects in the arguments. I understand that old objects may not be used in the function now, but having them and performing some sanity checks is generally recommended.

Reply:
@bsalamat 
It would be nice to know what sanity check should be performed.

thanks